tests/foreman/ui/test_ansible.py

```diff
@@ -565,6 +565,53 @@ def test_positive_ansible_config_report_changes_notice_and_failed_tasks_errors(
             assert f'notice Install the {package_name} package' in notice_log['permission_denied']
             assert f'Installed: rubygem-{package_name}' in notice_log['permission_denied']
 
+    def test_positive_non_admin_hostgroup_permission(
+        self, request, function_org, function_location, target_sat
+    ):
+        """Verify Hostgroup page is accessible to non-admin user.
+
+        :id: 3b8ce895-9d4c-491d-a5eb-d292748134a5
+
+        :customerscenario: true
+
+        :BZ: 2236418
+
+        :steps:
+            1. Create a user with non-admin permissions and assign "Ansible Roles Manager" and "Ansible Tower Inventory Reader"
```

Comment:
Checking [the code](https://github.com/theforeman/foreman_ansible/blob/master/lib/foreman_ansible/register.rb#L175-L178), it's the `Ansible Tower Inventory Reader` who provides the `view_hostgroups` permission.

It makes me think: Should we have some generic test that will accept an array of roles, in this case, "Ansible Roles Manager" and "Ansible Tower Inventory Reader," and then verify that with these roles, users can do whatever is part of the test, like here viewing the host group?

This problem can happen in every plugin that defines the roles, such as Katello, Remote Execution, Puppet, and so on.

Reply:
Looks like a valid test scenario. I will create an automation card for this and add it in backlog.

Reply:
Created a Jira for it: SAT-28233